dart/dynamics/BodyNode.cpp

```diff
@@ -888,34 +960,16 @@ void BodyNode::init(Skeleton* _skeleton)
 }
 
 //==============================================================================
-void BodyNode::draw(renderer::RenderInterface* _ri,
-                    const Eigen::Vector4d& _color,
-                    bool _useDefaultColor,
-                    int _depth) const
+void BodyNode::processNewEntity(Entity* _newChildEntity)
 {
-  if (_ri == NULL)
+  if(find(mChildBodyNodes.begin(), mChildBodyNodes.end(), _newChildEntity) !=
+     mChildBodyNodes.end())
```

Comment:
Do we need to check the duplicity here? It's already checked in `BodyNode::addChildBodyNode(~)` and also this function does not add the new entity to the list of child body nodes.